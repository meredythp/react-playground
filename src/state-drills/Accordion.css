button {
  padding: 3px;
  margin: 6px;
  border: 1px solid black;
  background-color: lightcoral;
}

ul {
  text-align: left;
  list-style: none;
}

.content {
  padding: 0 0 0 9px;
}

.none {
  display: none;
}

.block {
  display: block;
}