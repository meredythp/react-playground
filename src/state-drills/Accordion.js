import React from 'react';
import './Accordion.css'

class Accordion extends React.Component {
  static defaultProps = {
    sections: [],
  }

  state = {
    currentSectionIndex: null,
  };

  HandleButtonClick(index) {
    this.setState({
      currentSectionIndex: index,
    })    
  }

  /* DisplayContent(index) {
    let classState = index === this.state.currentSectionIndex ? "block" : "none";
    return classState
  } */

  Item(section, index) {
    return (
      <li key={index}>
        <button onClick={() => this.HandleButtonClick(index)}>
            {section.title}
        </button>
        {/* <div id="content" className={this.DisplayContent(index)}> */}
        {this.props.sections.length > 0 && index === this.state.currentSectionIndex && <div className="content"><p>{section.content}</p></div>}
        {/* </div> */}
      </li>
    )
  }

  render() {
    return (
      <div className="section-wrapper">
        <ul className="main-ul">
          {this.props.sections.map((section, index) => (
            this.Item(section, index)
          ))}
        </ul>
      </div>
    )
  }
}

export default Accordion;