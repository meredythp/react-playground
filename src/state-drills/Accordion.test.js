import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({
  adapter: new Adapter()
});

describe(`Accordion Component`, () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders no sections as active by default', () => {
    const tree = renderer
      .create(<Accordion />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders empty given no sections', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens a only the clicked section', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})