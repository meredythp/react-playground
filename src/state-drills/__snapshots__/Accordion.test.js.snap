// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component opens a clicked section 1`] = `
<div
  className="section-wrapper"
>
  <ul
    className="main-ul"
  >
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Section 1
      </button>
      <div
        className="content"
      >
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit.
        </p>
      </div>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Accordion Component renders empty given no sections 1`] = `
<div
  className="section-wrapper"
>
  <ul
    className="main-ul"
  />
</div>
`;

exports[`Accordion Component renders no sections as active by default 1`] = `
<div
  className="section-wrapper"
>
  <ul
    className="main-ul"
  />
</div>
`;
